src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4182,6 +4357,71 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
             return (method, results, shouldReturnNotNull);
         }
 
+        private void ApplyMemberPostConditions(MethodSymbol method)
+        {
+            if (method is null)
+            {
+                return;
+            }
+
+            do
+            {
+                var type = method.ContainingType;
+                var notNullMembers = method.NotNullMembers;
+                var notNullWhenTrueMembers = method.NotNullWhenTrueMembers;
+                var notNullWhenFalseMembers = method.NotNullWhenFalseMembers;
+
+                if (notNullWhenTrueMembers.IsEmpty && notNullWhenFalseMembers.IsEmpty)
+                {
+                    applyMemberPostConditions(type, notNullMembers, ref State);
```

Comment:
>applyMemberPostConditions [](start = 20, length = 25)

Are these not applied when there are notNullWhenTrue members?  Why not?  I would expect them to be imposed on both sides of a split state.

Reply:
Possibly move this line to just before the enclosing if

---
In reply to: [387995786](https://github.com/dotnet/roslyn/pull/41675#discussion_r387995786) [](ancestors = 387995786)